src/main/java/com/teragrep/pth10/ast/commands/evalstatement/EvalStatement.java

```diff
@@ -279,55 +174,34 @@ public Node visitL_evalStatement_evalLogicStatement(DPLParser.L_evalStatement_ev
     }
 
     private Node evalLogicStatementEmitCatalyst(DPLParser.L_evalStatement_evalLogicStatementContext ctx) {
-        Node rv = null;
-        Column lCol = null;
-        Column rCol = null;
+        final Column col;
         LOGGER
                 .debug(
                         "VisitEvalLogicStatement(Catalyst) incoming: children=<{}> text=<{}>", ctx.getChildCount(),
                         ctx.getText()
                 );
-        Node l = visit(ctx.getChild(0));
-        LOGGER.debug("VisitEvalLogicStatement(Catalyst) left: class=<{}>", l.getClass().getName());
-        if (ctx.getChildCount() == 1) {
-            // leaf
-            rv = l;
+
+        final List<DPLParser.EvalStatementContext> evalStatements = ctx.evalStatement();
+
+        if (evalStatements.size() != 2) {
```

Comment:
shouldn't this be restricted on pth_03 side that it enforces the correct amount of evalStatements?